cmd/gateway-gcs-layer.go

```diff
@@ -162,14 +183,14 @@ func newGCSGateway(endpoint string, projectID, secretKey string, secure bool) (G
 		return nil, err
 	}
 
-	anonClient, err := minio.NewCore("storage.googleapis.com", "", "", secure)
+	anonClient, err := minio.NewCore(endpoint, "", "", secure)
```

Comment:
Azure uses basic http client for anonymous operations, we should use that instead here as well. Using two SDKs for the GCS backend is not a good idea IMHO.